### ISSUE TITLE
RadioButtonGroup has no label and error label

### ISSUE THREAD

**Issue by lolytxd** [vaadin/vaadin-radio-button-flow#58]
Why there is no label and error label in this component?
I mean something like this:
![image](https://user-images.githubusercontent.com/11396254/41023158-e0b08944-696b-11e8-9e3a-41c51ecbeb36.png)

It should have setLabel method, to add label above radio buttons (eg. ```setLabel("Application type");```).
The same goes for error label below radio buttons.

When I bind RadioButtonGroup as required, text is not showing up anywhere.
```
RadioButtonGroup<ApplicationType> applicationType = new RadioButtonGroup<>();
applicationType.setItems(ApplicationType.values());
binder.forField(typWniosku).asRequired("You need to choose application type").bind(...);
binder.validate();
```

**Comment by lolytxd** [comment]
If someone wants a quick workaround:

```
public class RadioButtonGroupN<T> extends RadioButtonGroup<T> implements HasValidation {

	private CaptionLabel captionLabel = new CaptionLabel();
	private Label error = new Label();

	@Override
	public void setItems(Collection<T> items) {
		super.setItems(items);
		if (!getElement().getChild(0).equals(captionLabel.getElement())) {
			getElement().insertChild(0, captionLabel.getElement());
		}
	}

	@Override
	public void setErrorMessage(String errorMessage) {
		error.setText(errorMessage);
		error.addClassName("error-like");
		error.getStyle()
				.set("color", "var(--lumo-error-text-color)")
				.set("display", "block");
		add(error);
	}

	@Override
	public void setRequiredIndicatorVisible(boolean requiredIndicatorVisible) {
		if (captionLabel != null) {
			captionLabel.setRequiredIndicatorVisible(requiredIndicatorVisible);
		}
	}

	@Override
	public String getErrorMessage() {
		if (error != null) {
			return error.getText();
		}
		return null;
	}

	@Override
	public boolean isInvalid() {
		return error.isVisible();
	}

	@Override
	public void setInvalid(boolean invalid) {
		if (captionLabel != null) {
			captionLabel.setError(invalid);
		}
		error.setVisible(invalid);
	}

	public void setLabel(String label) {
		captionLabel.setText(label);
		captionLabel.getStyle().set("display", "table");
	}
}

class CaptionLabel extends Label {

	private boolean required;

	public CaptionLabel() {
		addClassName("caption-label");
		getStyle().set("white-space", "normal");
	}

	public void setLabel(String text) {
		setText(text);
	}

	public void setRequiredIndicatorVisible(boolean requiredIndicatorVisible) {
		required = requiredIndicatorVisible;
		if (requiredIndicatorVisible) {
			addClassName("caption-label-required");
		} else {
			removeClassName("caption-label-required");
		}
	}

	public void setError(boolean error) {
		if (error) {
			addClassName("caption-label-error");
			if (required) {
				addClassName("caption-label-required");
			}
		} else {
			removeClassName("caption-label-error");
			removeClassName("caption-label-required");
		}
	}

}
```

CSS styles:
```
		 .error-like {
		     margin-left: calc(var(--lumo-border-radius) / 4);
		     font-size: var(--lumo-font-size-xs);
		     line-height: var(--lumo-line-height-xs);
		     will-change: max-height;
		     transition: 0.4s max-height;
		     max-height: 5em;
		}
		 .caption-label {
		     padding-top: var(--lumo-space-m);
		     align-self: flex-start;
		     color: var(--lumo-secondary-text-color);
		     font-weight: 500;
		     font-size: var(--lumo-font-size-s);
		     margin-left: calc(var(--lumo-border-radius) / 4);
		     transition: color 0.2s;
		     line-height: 1;
		     padding-bottom: 0.5em;
		     overflow: hidden;
		     white-space: nowrap;
		     text-overflow: ellipsis;
		     position: relative;
		     max-width: 100%;
		     box-sizing: border-box;
		     padding-right: 1em;
		}
		 .caption-label::after {
		     content: var(--lumo-required-field-indicator, "•");
		     transition: opacity 0.2s;
		     opacity: 0;
		     color: var(--lumo-primary-text-color);
		     position: absolute;
		     right: 0;
		     width: 1em;
		     text-align: center;
		}
		 .caption-label-required::after {
		     opacity: 1;
		}
		 .caption-label-error::after {
		     color: var(--lumo-error-text-color);
		}
		
		.caption-label[disabled]{
			color: var(--lumo-disabled-text-color);
    		-webkit-text-fill-color: var(--lumo-disabled-text-color);
		}
```

**Comment by ZheSun88** [comment]
So far those two properties are not supported by the webcomonent itself, but there is already a ticket https://github.com/vaadin/vaadin-radio-button/issues/19 mentioning this.

**Comment by pleku** [comment]
This is duplicate of #12 in regards to error message and required indicator, but I'm keeping it open for the label part.

**Comment by ZheSun88** [comment]
The blocking ticket has been closed and Code has been generated with the latest version.

**Comment by ZheSun88** [comment]
This has been implemented with #69
